pyproject.toml

```
@@ -7,7 +7,7 @@ readme = "README.md"
 package-mode = false
 
 [tool.poetry.dependencies]
-python = "==3.12.2"
+python = "^3.12.2"
```

Let's see if making this explicit to `3.12.3` fixes the end-to-end tests:

```suggestion
python = "==3.12.3"
```

---

We're not going to do this just yet; waiting to see if getting the API updated is enough.